spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
From my point of view, there are a few advantages to GcpProjectIdProvider.
1) It reduces confusion by not requiring the user to get project ID from GcpProperties (and being confused with the pseudo-credentials in it),
2) It helps with testing, with the benefit that we don't need to inject/mock Strings but rather get something more strongly typed.

In any case, it seems better not to worry about this too much in this PR. I might file an issue to look into this later.